chain/stmgr/supply.go

```diff
@@ -198,24 +192,29 @@ func (sm *StateManager) setupPostCalicoVesting(ctx context.Context) error {
 func (sm *StateManager) GetFilVested(ctx context.Context, height abi.ChainEpoch) (abi.TokenAmount, error) {
 	vf := big.Zero()
 
-	sm.genesisMsigLk.Lock()
-	defer sm.genesisMsigLk.Unlock()
-
 	// TODO: combine all this?
-	if sm.preIgnitionVesting == nil || sm.genesisPledge.IsZero() || sm.genesisMarketFunds.IsZero() {
+	if sm.preIgnitionVesting == nil || sm.genesisPledge.IsZero() {
+		sm.genesisMsigLk.Lock()
```

Comment:
I think it might make more sense to move this lock into each of the vesting functions as they are also owned by the state manager and we are trying to make this lock more granular

Reply:
Sure, I'm happy to do that. Note that we _should_ only be calling each of these methods once, but I think that makes sense.